scripts/quests/otherAreas/Knocking_on_Forbidden_Doors.lua

```diff
@@ -0,0 +1,279 @@
+                onTrigger = function(player, npc)
+                    -- First time clicking on ???
+                    if player:hasKeyItem(xi.ki.MIRE_INCENSE) and
+                    quest:getVar(player, 'prog') == 2 then
+                        return quest:progressEvent(556, { [1] = xi.ki.MIRE_INCENSE })
+
+                    -- Second time clicking on ???
+                    elseif quest:getVar(player, 'prog') == 3 then
+                        return quest:progressEvent(557, { [1] = xi.ki.MIRE_INCENSE })
+
+                    -- Clicking on the ??? after killing NM
+                    elseif quest:getVar(player, 'prog') == 4 then
+                        return quest:event(558)
+                    end
+                end,
+            },
+
+            ['Alsha'] =
+            {
+                onMobDeath = function(mob, player, isKiller, firstCall)
+                    if quest:getVar(player, 'prog') == 3 and isKiller ~= mob then
+                        quest:setVar(player, 'prog', 4)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [556] = function (player, csid, option, npc)
+                    quest:setVar(player, 'prog', 3)
+                end,
+
+                [557] = function (player, csid, option, npc)
+                    if quest:getVar(player, 'prog') == 3 and
+                    npcUtil.popFromQM(player, npc, ID.mob.ALSHA, { claim = true, hide = 0 }) then
+                        return quest:messageSpecial(ID.text.FOUL_STENCH)
+                    end
+                end,
+
+                [558] = function(player, csid, option, npc)
+                    quest:setVar(player, 'prog', 5)
+                    player:addKeyItem(xi.ki.BETTER_HUMES_AND_MANNEQUINS)
+                    return quest:messageSpecial(ID.text.KEYITEM_OBTAINED, xi.ki.BETTER_HUMES_AND_MANNEQUINS)
+                end,
+            },
+        },
+
+        [xi.zone.MHAURA] =
+        {
+            ['Fyi_Chalmwoh'] =
+            {
+                onTrigger = function(player, csid, option, npc)
+                    if quest:getVar(player, 'prog') == 5 then
+                        return quest:progressEvent(321, { [0] = 704,
+                                                          [1] = 217,
+                                                          [2] = xi.mannequin.cost.PURCHASE,
+                                                          [3] = xi.mannequin.cost.TRADE,
+                                                          [4] = xi.mannequin.cost.POSE,
+                                                          [5] = player:getGil(),
+                                                          [6] = xi.ki.BETTER_HUMES_AND_MANNEQUINS })
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [321] = function(player, csid, option, npc)
+                    quest:complete(player)
```

Comment:
Can't forget to delete the temp key item from the player's inventory. Can also enclose it in if quest:complete
```suggestion
                    if quest:complete(player) then
                        player:delKeyItem(xi.ki.MIRE_INCENSE)
                    end
```